modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/CppRestClientCodegen.java

```diff
@@ -110,6 +112,10 @@ public CppRestClientCodegen() {
         addOption(DEFAULT_INCLUDE,
                 "The default include statement that should be placed in all headers for including things like the declspec (convention: #include \"Commons.h\" ",
                 this.defaultInclude);
+        addOption(GENERATE_INTERFACES_FOR_APIS,
+                "Generate abstract base classes (interfaces) for APIs. This allows to mocks APIs (e.g. for unit testing).");
```

Comment:
I wonder if we should simply force this, and not make it an option.
I agree that the gmock stuff should be optional because we don't want to force users into gmock.
But having interfaces shouldn't impact anyone already using the tool, and "forcing" them on everyone will help make the overall project more maintainable (e.g. if one day the sample code is tested, should there be tests for both versions?).

Reply:
My reasoning was that having the interfaces will increase compile and runtime costs (e.g. an additional virtual function call). So if you don't need them, why would you want to pay for them? On the other hand, compared to a HTTP request, a virtual function call is negligible so this argument is not really valid. 
Your argument regarding maintainability is certainly right and probably the stronger one, especially for an open source project. So I will remove this config option.